packages/cli/README.md

```diff
@@ -127,3 +127,615 @@ jobs:
+    const filePath = join(directory, file);
+    const stats = await stat(filePath);
+
+    // if file, get file size info
+    if (stats.isFile()) {
+      results.push({ filePath, size: stats.size });
+    }
+    // if directory, recurse
+    else if (stats.isDirectory()) {
+      results.push(...(await getFileSizeData({ directory: filePath })));
+    }
+  }
+
+  return results;
+}
+```
+
+The above code will recursively get the file size info for all files in the specified directory and it's subdirectories.
+
+Let's create a runner function that uses the above code to get the file size data and turns it into the shape
+of `AuditOutputs`:
+
+**file-size `RunnerFunction`**
+
+```typescript
+// code-pushup.config.ts
+import { pluralize } from '@code-pushup/utils';
+
+// ...
+
+async function runnerFunction(options: Options): Promise<AuditOutputs> {
+  const data = await getFileSizeData(options);
+  // We use a helper function of the utils package to pluralize the display value
+  const displayValue = `${data.length} ${data.length === 1 ? 'file' : pluralize('file')}`;
+
+  let fileSizeAuditOutput: AuditOutput = {
+    slug: fileSizeAudit.slug,
+    // We have always a score of 1. Proper implementation of scores will be covered in a next section.
+    score: 1,
+    value: data.length,
+    displayValue,
+  };
+
+  return [fileSizeAuditOutput];
+}
+```
+
+Now we can execute the CLI with `npx code-pushup collect --no-progress` and see a similar output as the following:
+
+**stdout of basic file-size plugin**
+
+```sh
+Code PushUp Report - @code-pushup/core@x.y.z
+
+File size plugin audits
+● File size audit                                                            2
+
+Categories
+┌──────────┬───────┬────────┐
+│ Category │ Score │ Audits │
+└──────────┴───────┴────────┘
+
+Made with ❤ by code-pushup.dev
+```
+
+### RunnerConfig
+
+The second way to write a plugin runner is a `RunnerConfig`.
+This option is less flexible but can be used in cases when you have to use another CLI.
+why runner function can't be used...
+
+We will implement a performance focused plugin using the [Lighthouse CLI](https://github.com/GoogleChrome/lighthouse#using-the-node-cli) as real life example.
+
+Let's start with a `crate` function maintaining the basic information of the `PluginConfig`.
+
+<details>
+<summary> <b>lighthouse plugin setup</b> (collapsed for brevity) </summary>
+
+```typescript
+// code-pushup.config.ts
+import { PluginConfig, RunnerConfig } from '@code-pushup/models';
+
+type Options = {
+  url: string;
+  onlyAudits: string;
+};
+
+async function create(options: Options): Promise<PluginConfig> {
+  const lcpAudit = {
+    slug: 'largest-contentful-paint',
+    title: 'Largest Contnentful Paint',
+  };
+
+  return {
+    slug: 'lighthouse',
+    title: 'Chrome Lighthouse plugin',
+    icon: 'lighthouse',
+    audits: [
+      lcpAudit,
+      // other lighthouse audits
+    ],
+    runner: runnerConfig(options),
+  };
+}
```

Comment:
Both the `create` function examples return a promise only because of the `async` keyword (no `await` in the body), which necessitates using `await create(...)` in the config. Wouldn't it be simpler to have sync `create` functions instead of having them be artificially async? On the other hand, it's good mention we how to do async create functions somewhere (though perhaps seeing `await eslintPlugin(...)` in our examples is enough of an indication). :thinking: